src/Chainweb/Miner/Coordinator.hs

```diff
@@ -167,6 +164,54 @@ filterAdjustments newBh as = case window $ _blockChainwebVersion newBh of
             limit = bool (_blockHeight newBh - wh) 0 (_blockHeight newBh < wh)
         in HM.filter (\(T2 h _) -> h > limit) as
 
+-- | THREAD: Accepts a "solved" `BlockHeader` from some external source (likely
+-- a remote mining client), reassociates it with the `Cut` from which it
+-- originated, and publishes it to the `Cut` network.
+--
+publishing :: LogFunction -> TVar (Maybe MiningState) -> CutDb cas -> BlockHeader -> IO ()
+publishing lf tp cdb bh = do
+    -- Reassociate the new `BlockHeader` with the current `Cut`, if possible.
+    -- Otherwise, return silently.
+    --
+    c <- _cut cdb
+    readTVarIO tp >>= \case
+        Nothing -> pure ()  -- TODO Throw error?
+        Just (MiningState pl p) -> when (compatibleCut c && samePayload pl) $ do
+            -- Publish the new Cut into the CutDb (add to queue).
+            --
+            c' <- monotonicCutExtension c bh
```

Comment:
Tests with 4dcd935 indicate that about 15% of all mined cuts are mined on outdated cuts. 

There were between 4 and 5 cases per minute in the network at an overall block rate of 2.5 blocks per chain per minute for 10 chains.

That number seems a bit high, because it should depend only on local latencies and not on p2p network latencies.

Reply:
And these are running in "in-process" mode - the remote miner isn't involved.